### ISSUE TITLE
Elasticsearch 1.* upwards does not seem to work

### ISSUE THREAD

**Issue by thaisonle94** [matteodem/meteor-easy-search#115]
I have used your **meteor-easy-search** package to create a demo project using **elastic-search** for its search feature.
This is my [git project repo](https://github.com/thaisonle94/meteor-elasticsearch-demo). Before I applied 'elastic-search' to my project, it runs just fine with the following EasySearch settings:  
```javascript
EasySearch.createSearchIndex('index-players', {
    'collection': Players, // instanceof Meteor.Collection
    'field': ['name', 'score', 'category'], // array of fields to be searchable
    'limit': 10,
    'use' : 'mongo-db',
    'convertNumbers': true,
    'sort': function() {
        return { 'score': -1, 'name': -1 };
    }
});
```
But after I applied 'elastic-search' by following [these instructions](http://matteodem.github.io/meteor-easy-search/docs/elastic-search/), my app cannot search any more. *(and I have already started the ElasticSearch service using `sudo service elasticsearch start`). When I run `curl http://localhost:9200` it does return an object.* This is my current EasySearch settings:  
```javascript
'collection': Players, // instanceof Meteor.Collection
    'field': ['name', 'score', 'category'], // array of fields to be searchable
    'limit': 10,
    'use': 'elastic-search',
    'query': function (searchString) {
        var query = {
            "fuzzy_like_this": {
                "fields": ["name", "category"],
                "like_text": searchString
            }
        };
        return query;
    },
    'sort': function () {
        return {
            "score": {"order": "desc"}
        };
    }
```
Now when I type in the keywords for searching, it doesn't return anything, and the weird thing is the `Load More` button still show up.  
What get me stuck is that I don't get any errors neither on the client browser console nor the server console which *may* help me find the way to fix this. So I hope you can pay a little of your time to take a look at [my project](https://github.com/thaisonle94/meteor-elasticsearch-demo) and help me fixing this problem.
Thanks a lot in advanced !

**Comment by matteodem** [comment]
L31 should be 

```html
{{#esEach index="index-players"}}
...
{{/esEach}}
```

instead of https://github.com/thaisonle94/meteor-elasticsearch-demo/blob/master/client/pages/leaderboard.html#L31

**Comment by thaisonle94** [comment]
I've edited as your above comment but it seems like nothing's changed. I think it's not the reason for my problem, because as I mentioned before, my project runs just fine (the **search** feature fully works!) with the EasySearch settings which uses `'mongo-db'` instead of `'elastic-search'`.

**Comment by matteodem** [comment]
I will have a look at your repo later

**Comment by thaisonle94** [comment]
Thanks a lot :) ! I'm looking forward to hearing from you !

**Comment by matteodem** [comment]
I might know the problem, you need to remove "score" as a field, since fuzzy like this queries cannot handle numeric values.

**Comment by thaisonle94** [comment]
So I removed `score` as a field but the problem still appears. This is the modified code in my **easy-search.js** file:
```javascript
EasySearch.createSearchIndex('index-players', {
    'collection': Players, // instanceof Meteor.Collection
    'field': ['name', 'category'], // array of fields to be searchable
    'limit': 10,
    'use': 'elastic-search',
    'query': function (searchString) {
        var query = {
            "fuzzy_like_this": {
                "fields": ["name", "category"],
                "like_text": searchString
            }
        };
        return query;
    },
    'sort': function () {
        return {
            "score": {"order": "desc"}
        };
    }
});
```
The weird thing is that: when I come back to use `'mongo-db'` search engine, the instant feature works just fine. This is the code when I switch to `'mongo-db'`:
```javascript
EasySearch.createSearchIndex('index-players', {
    'collection': Players, // instanceof Meteor.Collection
    'field': ['name', 'category'], // array of fields to be searchable
    'limit': 10,
    'use' : 'mongo-db',
    'sort': function() {
        return { 'score': -1, 'name': -1 };
    }
});
```
I just don't know how the app behaves weirdly when I set `'elastic-search'` as my search engine: When I type in the keywords, it shows: **No results** but the `Load More` button still show up. (You can see this problem when you clone and test [my repo](https://github.com/thaisonle94/meteor-elasticsearch-demo)).

**Comment by matteodem** [comment]
I finally had time to clone this repo and I started my elasticsearch through [brew](http://brew.sh/) and it worked: http://screencast.com/t/6yBJH56BgS

**Comment by thaisonle94** [comment]
Woao :dizzy_face: I didn't expected this. I'm running `Ubuntu 14.04` with `elasticsearch v1.4.2` run on port `9200`. Can you tell me which version of *elasticsearch* that you're using ?

**Comment by matteodem** [comment]
```shell
[2015-01-16 23:35:16,096][INFO ][node                     ] [Archer] version[0.90.3], pid[34135], build[5c38d60/2013-08-06T13:18:31Z]
```

On a Mac with Yosemite 10.10.1

**Comment by thaisonle94** [comment]
My app still doesn't work. This is what `elasticsearch` returns when I go for `curl localhost:9200`:
![all text](http://s14.postimg.org/cbrwgqdw1/meteor_easy_search_feedback1.png "meteor-elastic-search-feedback")
 I've tested in one of my co-workers notebook who is running `Ubuntu 14.04` too and got the same results. So I guess... Is this the compatible problem of your package with my Ubuntu version (**14.04**) or my elasticsearch version (**1.4.2**) ?

**Comment by matteodem** [comment]
I just updated my local elasticsearch to 1.0.1 and it does not work anymore too. I'll have a closer look in the next days.

**Comment by matteodem** [comment]
Please update again to the latest version and see if it works now.

**Comment by thaisonle94** [comment]
I've just updated my project to `Meteor 1.0.3.1` and `matteodem:easy-search 1.4.9`. (with the same old code, which runs well if I set to `'use' : 'mongo-db'`). Now my app still behaves unexpectedly:
![](http://s18.postimg.org/va6iv3yll/meteor_easy_search.png)  
- The input and the result seems not match
- there's only about 1-2 results for each of my search attempts with a random pre-provided name. Some of them, as mentioned above, doesn't event match with the query. And despite having only 1-2 results, the `Load More` button still appear and each time I click at that button, it show me 2 more results. This is weird because I've set my default limit & pagination to `10` docs.

**Comment by matteodem** [comment]
Yes, it is also behaving very unexpected on my machine. It is weird since it finds all the id's but the mapping is not happening right. I will have a closer look soonish.

**Comment by thaisonle94** [comment]
Thanks ! I'm looking forward to hearing from you :)

**Comment by matteodem** [comment]
I had a look at this and it seems like a really weird error. It seems like Elastic-Search generates weird id's, even though I specify them to be the mongo-db document id. Then when mapping the elastic-search ids back to mongo docs they are not found.

**Comment by thaisonle94** [comment]
Yeah, it's really weird ... It seems like a major error, I hope you figure out the solution and fix it soon ! :)

**Comment by matteodem** [comment]
I know what the problem was, the Elasticsearch Index had too many leftover docs in it. Can you please try to delete index-players and add the docs again? It should work.

**Comment by matteodem** [comment]
Feel free to write here if it did not fix the problem.

**Comment by thaisonle94** [comment]
What do you mean by "deleting index-players and adding the docs again" ? I tried re-install `ElasticSearch` to **v1.4.2**, upgrade your package to **v1.4.10**, run `meteor reset` but it doesn't seem to work.

**Comment by matteodem** [comment]
Simply removing all the player elastic search docs, and re-indexing / re-inserting the mongo docs, which also re-indexes the ES docs.

**Comment by kyooriouskoala** [comment]
Hi, I'm having the same issue that @thaisonle94 experienced before. The issue happened when I started to delete all the data inside the collection for the second time. 

The first time I deleted all the data was to insert new data format (different fields) and it worked fine. 
The second time I deleted it all and use your demo players data format again it stopped working properly.

The result shows nothing found but the pagination numbers appear. I also tried thaisonle94's repo which worked find when I loaded it up for the first time. However, when I deleted the data for the second time to insert back new data, it just stopped working properly too.

Apart from removing the indexes and emptying the collection on MongoDB as well as doing `meteor reset`, I also deleted the ElasticSearch [indexes] (https://www.elastic.co/guide/en/elasticsearch/reference/1.6/indices-delete-index.html) and cleared [caches] (https://www.elastic.co/guide/en/elasticsearch/reference/1.x/indices-clearcache.html); just in case. Nothing works again still 😞

Much appreciate your support in looking into this bug.
